### ISSUE TITLE
feat[dev]: ✨ Accessible issue found in sd-input

### ISSUE THREAD

**Issue by MartaPintoTeixeira** [solid-design-system/solid#2237]
## User Story
While testing sd-input I have noticed we might have an accessibility issue with the sample Spin Buttons.

Issue:
When I mouse hover the sd-input when minus icon is grey it is not accessible.
When icon is not active it should have the same color are the closable icon.

<img width="1069" alt="Image" src="https://github.com/user-attachments/assets/99d4b5ed-d2ba-480a-b624-a76afebe5a08" />
<img width="565" alt="Image" src="https://github.com/user-attachments/assets/80a80e7a-4d2d-47b8-83ed-4ff783557e5e" />

### Suggested Solution
Use same color as clearable icon - neutral-800

## DoR
- [ ] Item has business value
- [ ] Item has been estimated by the team
- [ ] Item is clear and well-defined
- [ ] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
